src/libraries/System.Net.Mail/tests/Functional/MockSmtpServer.cs

```diff
@@ -0,0 +1,284 @@
+        }
+
+        private async Task HandleConnectionAsync(Socket socket)
+        {
+            var buffer = new byte[1024].AsMemory();
+
+            async ValueTask<string> ReceiveMessageAsync(bool isBody = false)
+            {
+                var terminator = isBody ? BodyTerminator : MessageTerminator;
+                int suffix = terminator.Length;
+
+                int received = 0;
+                do
+                {
+                    int read = await socket.ReceiveAsync(buffer.Slice(received), SocketFlags.None);
+                    if (read == 0) return null;
+                    received += read;
+                }
+                while (received < suffix || !buffer.Slice(received - suffix, suffix).Span.SequenceEqual(terminator.Span));
+
+                MessagesReceived++;
+                return Encoding.UTF8.GetString(buffer.Span.Slice(0, received - suffix));
+            }
+            async ValueTask SendMessageAsync(string text)
+            {
+                var bytes = buffer.Slice(0, Encoding.UTF8.GetBytes(text, buffer.Span) + 2);
+                bytes.Span[^2] = (byte)'\r';
+                bytes.Span[^1] = (byte)'\n';
+                await socket.SendAsync(bytes, SocketFlags.None);
+            }
+
+            try
+            {
+                OnConnected?.Invoke(socket);
+                await SendMessageAsync("220 localhost");
+
+                string message = await ReceiveMessageAsync();
+                Debug.Assert(message.ToLower().StartsWith("helo ") || message.ToLower().StartsWith("ehlo "));
+                ClientDomain = message.Substring(5).ToLower();
+                OnCommandReceived?.Invoke(message.Substring(0, 4), ClientDomain);
+                OnHelloReceived?.Invoke(ClientDomain);
+
+                await SendMessageAsync("250-localhost, mock server here");
+                if (SupportSmtpUTF8) await SendMessageAsync("250-SMTPUTF8");
+                await SendMessageAsync("250 AUTH PLAIN LOGIN" + (AdvertiseNtlmAuthSupport ? " NTLM" : ""));
+
+                while ((message = await ReceiveMessageAsync()) != null)
+                {
+                    int colonIndex = message.IndexOf(':');
+                    string command = colonIndex == -1 ? message : message.Substring(0, colonIndex);
+                    string argument = command.Length == message.Length ? string.Empty : message.Substring(colonIndex + 1).Trim();
+
+                    OnCommandReceived?.Invoke(command, argument);
+
+                    if (command.StartsWith("AUTH", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var parts = command.Split(' ');
+                        Debug.Assert(parts.Length > 1, "Expected an actual auth request");
+
+                        AuthMethodUsed = parts[1];
+
+                        // PLAIN is not supported by SmtpClient
+                        /*
+                        if (parts[1].Equals("PLAIN", StringComparison.OrdinalIgnoreCase))
```

Comment:
Couldn't find anything explicit in the coding guidelines, but I think we should avoid committing commented out code.

Reply:
Correct. We never checked in commented out code.

Reply:
Oops :)
Will remove it